napari/utils/events/_tests/test_evented_model.py

```diff
@@ -103,6 +104,86 @@ class Model(EventedModel):
         model.shaped2_values = [1]
 
 
+def test_evented_model_array_updates():
+    """Test updating an evented pydantic model with an array."""
+
+    class Model(EventedModel):
+        """Demo evented model."""
+
+        values: Array[int]
+
+    model = Model(values=[1, 2, 3])
+
+    # Mock events
+    model.events.values = Mock(model.events.values)
+
+    np.testing.assert_almost_equal(model.values, np.array([1, 2, 3]))
+
+    # Updating with new data
+    model.values = [1, 2, 4]
+    assert model.events.values.call_count == 1
+    np.testing.assert_almost_equal(
+        model.events.values.call_args[1]['value'], np.array([1, 2, 4])
+    )
+    model.events.values.reset_mock()
+
+    # Updating with same data, no event should be emitted
+    model.values = [1, 2, 4]
+    model.events.values.assert_not_called()
+
+
+def test_evented_model_array_equality():
+    """Test checking equality with an evented model with custom array."""
+
+    class Model(EventedModel):
+        """Demo evented model."""
+
+        values: Array[int]
```

Comment:
Just a note looking at these tests we should expand our `Array` so that if someone passes a dask array we don't coerce to numpy but leave as dask arrays, that can come later though